source/building-your-own-javascript-components/index.html.md.erb

```diff
@@ -0,0 +1,283 @@
+---
+title: Building your own JavaScript components
+weight: 75
+---
+
+# Building your own JavaScript components
+
+To help you build and initialise your own JavaScript components, GOV.UK Frontend provides some of its internal features for you to reuse:
+
+- a [`Component` class](#using-the-component-class) that handles common logistics across components
+- a [`createAll` function](#using-createall-with-your-components) that can initialise both GOV.UK Frontend and your own components
+- an [`isSupported` function](#checking-for-gov-uk-frontend-support-with-issupported) that checks if GOV.UK Frontend is supported
+
+Using GOV.UK Frontend’s features means you will not have to redevelop these features yourself, and it reduces the size of the [bundled JavaScript](../import-javascript/#import-javascript-using-a-bundler).
+
+## Using the `Component` class
+
+JavaScript components in GOV.UK Frontend share a number of behaviours. They:
+
+- check that GOV.UK Frontend is supported
+- check that the component is not already initialised on its root element
+- store the root element of the component
+
+The `Component` class implements these behaviours in an extensible way so you can focus the code of your component on its specific behaviour.
+
+### Defining static properties on components
+
+Some of the `Component`’s features are supported by static properties on the component inheriting from it. However, native static properties are not a JavaScript syntax supported by all [browsers that support `<script type=”module”>`](../import-javascript/#import-javascript).
```

Comment:
The gerund ('-ing') seems a bit confusing, this might read better if it's accurate.

```suggestion
Some of the `Component`’s features are supported by static properties on the component that inherit from it. However, native static properties are not a JavaScript syntax supported by all [browsers that support `<script type=”module”>`](../import-javascript/#import-javascript).
```